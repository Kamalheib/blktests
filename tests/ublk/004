#!/bin/bash
# SPDX-License-Identifier: GPL-3.0+
# Copyright (C) 2023 Ziyang Zhang
#
# Test ublk crash with delete just after daemon kill

. tests/ublk/rc

DESCRIPTION="test ublk crash with delete just after daemon kill"

_run() {
	local type=$1

	if [ "$type" == "null" ]; then
		${UBLK_PROG} add -t null -n 0 > "$FULL" 2>&1
	else
		truncate -s 1G "$TMPDIR/img"
		${UBLK_PROG} add -t loop -f "$TMPDIR/img" -n 0 > "$FULL" 2>&1
	fi

	udevadm settle
	if ! ${UBLK_PROG} list -n 0 >> "$FULL" 2>&1; then
		echo "fail to list dev"
	fi

	_run_fio_rand_io --filename=/dev/ublkb0 --time_based \
			--runtime=30 >> "$FULL" 2>&1 &
	sleep 2

	kill -9 "$(_get_ublk_daemon_pid 0)"

	${UBLK_PROG} del -n 0 >> "$FULL" 2>&1
}

test() {
	echo "Running ${TEST_NAME}"

	if ! _init_ublk; then
		return 1
	fi

	for type in "null" "loop"; do
		_run "$type"
	done

	_exit_ublk

	echo "Test complete"
}
