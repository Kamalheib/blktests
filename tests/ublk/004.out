Running ublk/004
Test complete
